pkg/utils/ldap/identity.go

```diff
@@ -151,6 +155,22 @@ func (i *Identity) Setup() error {
 		return fmt.Errorf("error configuring group substring filter type: %w", err)
 	}
 
+	dm := i.User.DisableMechanism
+	if dm == "" || dm == "none" || dm == "attribute" || dm == "group" {
+		if dm == "attribute" || dm == "group" {
+			if i.User.EnabledProperty == "" {
```

Comment:
I think we need to support the case we `dm == "group"` and `i.User.EnabledProperty==""`. I.e. only evaluating the enabled state based on the group membership.

Reply:
BTW, I think this would more readable if it used a `switch` statement instead of having those nested `if`s with lots of conditions.